xmtp_mls/src/utils/test/mod.rs

```diff
@@ -373,3 +377,37 @@ pub fn get_test_mode_malformed_installations() -> Vec<Vec<u8>> {
         })
         .collect()
 }
+
+#[cfg(any(test, feature = "test-utils"))]
+#[warn(dead_code)]
+/// Sets test mode to mimic future wrong epoch state.
+pub fn set_test_mode_future_wrong_epoch(enable: bool) {
+    use std::env;
+    if enable {
+        env::set_var("TEST_MODE_FUTURE_WRONG_EPOCH", "true");
+    } else {
+        env::set_var("TEST_MODE_FUTURE_WRONG_EPOCH", "false");
+    }
+}
+#[cfg(any(test, feature = "test-utils"))]
+/// Checks if test mode is enabled.
+pub fn is_test_mode_future_wrong_epoch() -> bool {
+    use std::env;
+    env::var("TEST_MODE_FUTURE_WRONG_EPOCH").unwrap_or_else(|_| "false".to_string()) == "true"
+}
+
+pub fn maybe_mock_wrong_epoch_for_tests() -> Result<(), GroupMessageProcessingError> {
+    if is_test_mode_future_wrong_epoch() {
+        return Err(OpenMlsProcessMessage(ProcessMessageError::ValidationError(
+            WrongEpoch,
+        )));
+    }
+    Ok(())
+}
+
+pub fn maybe_mock_future_epoch_for_tests() -> Result<(), GroupMessageProcessingError> {
```

Comment:
These two seem very similar - is there a reason why we need to mock them in two different locations? Also, are we no longer handling the scenario where a message is in the right epoch, but undecryptable?

Reply:
Yes, but they return two different errors for two different places, one is when we process a message and just get a wrong epoch, and one in upper layers where we get future epoch.